plugin/hooks.go

```diff
@@ -133,6 +134,11 @@ type Hooks interface {
 	// UserHasLoggedIn is invoked after a user has logged in.
 	UserHasLoggedIn(c *Context, user *model.User)
 
+	// UserWasCreated is invoked after a user was created.
+	//
+	// Minimum server version: 5.10
+	UserWasCreated(c *Context, user *model.User)
```

Comment:
Can you move this up before the `UserWillLogIn` hook.

Reply:
I'm proposing to move `UserWillLogIn`/`UserHasBeenCreated` before `UserWillLogIn`. You don't have to move `UserWillLogIn` for this. My thinking is, that having the create-hook before the login-hook follows the natural flow of a user.
Does this makes sense to you?

Reply:
I had in mind moving the `UserWasCreated`/`UserHasBeenCreated`, not `UserWillLogIn`. Sorry for the mistake. 

Yes, it makes sense to me. Thanks for the explanation. But what do you think about moving the `UserWasCreated`/`UserHasBeenCreated` up even before the `UserHasJoinedChannel`? It should follow the natural flow of a user because the user should be created before he can join a channel.

Reply:
Lets go even further and move `UserHasBeenCreated` and `UserWill/HasLogIn` before `MessageWillBePosted`.

Reply:
Done.